ui/hooks/usePolling.ts

```diff
@@ -28,43 +27,45 @@ const usePolling = <PollingInput>(
   }, []);
 
   useEffect(() => {
-    if (usePollingOptions.enabled === false || !hasPollingInputChanged) {
-      return () => {
-        // noop
-      };
-    }
+    callIdRef.current += 1;
+    const currentCallId = callIdRef.current;
 
     const cleanup = () => {
       if (pollTokenRef.current) {
         usePollingOptions.stopPollingByPollingToken(pollTokenRef.current);
-        cleanupRef.current?.(pollTokenRef.current);
+        pollTokenRef.current = null;
       }
     };
 
+    if (usePollingOptions.enabled === false) {
+      cleanup();
+      return () => {
+        // no-op
+      };
```

Comment:
Nit: We should be able to return nothing here rather than an empty function

```suggestion
      return undefined;
```

Reply:
That runs afoul of ESLint😂. If a cleanup function is used at all, then early exits need to return noops as well.
```
Arrow function expected no return value.eslint[consistent-return](https://eslint.org/docs/latest/rules/consistent-return)
```

Reply:
`consistent-return` just demands that we return _something_. It accepts `undefined` as a return value.

It is a bit silly for TypeScript code to be honest 🤔 Maybe we should scrap `consistent-return` for TypeScript files. The mistake it intends to guard against (accidentally returning nothing when a value is expected) is prevented by the return type

Reply:
Oh you're right! Agreed that the rule doesn't help for TypeScript files (can't think of a situation where it would add value). It's also actively confusing. 

`@metamask/eslint-config-typescript` would be the best place to fix it but we're 6 major versions behind so it would take forever to arrive.. I'll create a new PR doing this:

https://github.com/MetaMask/metamask-extension/blob/8223358a6a76342ba576047749439d7a5d8f660c/.eslintrc.js#L315-L322

```diff
diff --git a/.eslintrc.js b/.eslintrc.js
index bba1392e6d..fe0014a02c 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -342,6 +342,7 @@ module.exports = {
         'react/no-children-prop': 'off',
         'react/jsx-key': 'warn', // TODO - increase this into 'error' level
         'react-hooks/rules-of-hooks': 'error',
+        'consistent-return': 'off',
       },
       settings: {
         react: {
diff --git a/ui/hooks/usePolling.ts b/ui/hooks/usePolling.ts
index a535bd8834..70963d97a5 100644
--- a/ui/hooks/usePolling.ts
+++ b/ui/hooks/usePolling.ts
@@ -39,9 +39,7 @@ const usePolling = <PollingInput extends Json>(
 
     if (usePollingOptions.enabled === false) {
       cleanup();
-      return () => {
-        // no-op
-      };
+      return;
     }
 
     // Start polling when the component mounts
```